R/fread.R

```diff
@@ -311,7 +311,7 @@ yaml=FALSE, autostart=NA, tmpdir=tempdir(), tz="UTC")
              methods::as(v, new_class))
       },
       warning = fun <- function(e) {
-        warningf("Column '%s' was requested to be '%s' but fread encountered the following %s:\n\t%s\nso the column has been left as type '%s'", names(ans)[j], new_class, if (inherits(e, "error")) "error" else "warning", e$message, typeof(v))
+        warningf("Column '%s' was requested to be '%s' but fread encountered the following %s:\n\t%s\nso the column has been left as type '%s'", names(ans)[j], new_class, if (inherits(e, "error")) gettext("error") else gettext("warning"), e$message, typeof(v))
```

Comment:
```suggestion
        warningf(if (inherits(e, "error")) gettext("Column '%s' was requested to be '%s' but fread encountered the following error:\n\t%s\nso the column has been left as type '%s'") else gettext("Column '%s' was requested to be '%s' but fread encountered the following warning:\n\t%s\nso the column has been left as type '%s'"), names(ans)[j], new_class, e$message, typeof(v))
```
Can we branch this too? This will also help with inflection, both due to grammatical case (nominative "ошибк**а**" but "encountered" "ошибк**у**", accusative) and grammatical gender ("данн**ое** предупреждение", warning, but "данн**ую** ошибку", error)

Reply:
OK, I moved it outside the `warningf()` call though for readability.